update/Pos/Update/Download.hs

```diff
@@ -170,11 +174,13 @@ downloadHash ::
 downloadHash updateServers h = do
     manager <- liftIO $ newManager tlsManagerSettings
 
-    let -- try all servers in turn until there's a Right
+    let
+        maxLatency = 30 :: Minute
+        -- try all servers in turn until there's a Right
         go errs (serv:rest) = do
             let uri = toString serv <//> showHash h
             logDebug $ "Trying url " <> show uri
-            liftIO (downloadUri manager uri h) >>= \case
+            liftIO (withTimeOut ((fromInteger . toMicroseconds) maxLatency) (downloadUri manager uri h)) >>= \case
```

Comment:
It would be even better to pass `Minute` to `withTimeOut` and convert it to `Int` there. Also you can put `maxLatency` inside `where` block and then you won't need to pass this value (but that's subjective).